File: ruby/three.rb
def reverse(x)
    neg = x < 0
    rev = 0
    if neg
        x = -x
    end
    while x > 0
        rev = rev * 10 + x % 10
        x = x / 10
    end
    if neg
        return -rev
    else 
        return rev
    end
end    
print("Введите количество чисел: ")
n = gets.chomp().to_i
i = 0
while i < n
    print("Введите число: ")
    num = gets.chomp().to_i
    puts reverse(num)
    i += 1
end

File: Java/three.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Введите количество чисел: ");
        int n = scanner.nextInt();
        for (int i = 0; i < n; i++) {
            System.out.print("Введите число: ");
            int num = scanner.nextInt();
            System.out.println(reverse(num));
        }
    }
    public static int reverse(int x) {
        boolean negative = x < 0;
        int rev = 0;
        if (negative) {
            x = -x;
        }
        while (x > 0) {
            rev = rev * 10 + x % 10;
            x = x / 10;
        }
        return negative ? -rev : rev;
    }
}
